Commands/Public/roleinfo.js

```diff
@@ -0,0 +1,93 @@
+const moment = require("moment");
+const PaginatedEmbed = require("../../Modules/MessageUtils/PaginatedEmbed");
+const Permissions = require("discord.js/src/util/Permissions.js");
+
+const rolesPerPage = 25;
+
+module.exports = async ({ client, Constants: { Colors, Text }, Utils: { TitlecasePermissions } }, documents, msg, commandData) => {
+	if (!msg.suffix) {
+		const guildRoles = [...msg.guild.roles.values()].sort((a, b) => b.position - a.position);
+		const descriptions = [];
+		for (let i = 0; i < guildRoles.length; i += rolesPerPage) {
+			const roleSegment = guildRoles.slice(i, i + rolesPerPage).join("\n");
+			descriptions.push(`These are the roles on this server:\n\n${i ? `...${i} previous roles\n` : ""}${roleSegment}${i + rolesPerPage < guildRoles.length ? `\n...and ${guildRoles.length - i - rolesPerPage} more` : ""}`);
+		}
+
+		const memberRoles = [...msg.member.roles.values()].sort((a, b) => b.position - a.position);
+		const totalPermissionsBitfield = memberRoles.reduce((a, b) => a | b.permissions, 0); // eslint-disable-line no-bitwise
+		const totalPermissions = new Permissions(totalPermissionsBitfield);
+		for (let i = 0; i < memberRoles.length; i += rolesPerPage) {
+			const roleSegment = memberRoles.slice(i, i + rolesPerPage).join("\n");
+			descriptions.push([
+				`You currently have those roles:\n\n${i ? `...${i} previous roles\n` : ""}${roleSegment}${i + rolesPerPage < memberRoles.length ? `\n...and ${memberRoles.length - i - rolesPerPage} more` : ""}`,
+				"\nYour roles grant you get the following permissions:",
+				totalPermissionsBitfield ? `\`\`\`${TitlecasePermissions(totalPermissions.toArray(false).join(", "))}\`\`\`` : "You do not have any permissions on this server ⁉️",
+				totalPermissions.has(Permissions.FLAGS.ADMINISTRATOR, false) ? "⚠️ You have Administrator permissions which bypass any other permission or override" : "",
+			].join("\n"));
+		}
+		if (descriptions.length === 2 && descriptions[0].length + descriptions[1].length < 2048) {
+			return msg.send({
+				embed: {
+					color: Colors.INFO,
+					title: `This guild has ${guildRoles.length} roles:`,
+					description: `${descriptions[0]}\n\n${descriptions[1]}`,
+				},
+			});
+		}
+
+		await new PaginatedEmbed(msg, {
+			color: Colors.INFO,
+			title: `This guild has ${guildRoles.length} roles`,
+			description: `{description}`,
+			footer: `Page {currentPage} out of {totalPages}`,
+		}, {
+			descriptions,
+		}).init();
+	} else {
+		let role;
+		try {
+			role = await client.roleSearch(msg.suffix, msg.guild);
+		} catch (err) {
+			if (err.code === "FAILED_TO_FIND") {
+				return msg.send({
+					embed: {
+						color: Colors.SOFT_ERR,
+						title: "What's that role? 🏷️",
+						description: "I do not know anything about a role with that name.",
+						footer: {
+							text: "You can find a list of all roles by simply executing this command without any arguments.",
+						},
+					},
+				});
+			}
+			throw err;
```

Comment:
Error should not be thrown if soft. (Soft errors are errors that we can handle user-friendly, like FAILED_TO_FIND)

Reply:
Perhaps wrap it in an `else {}`?